mlflow/tracing/trace_manager.py

```diff
@@ -0,0 +1,177 @@
+import logging
+import threading
+from dataclasses import dataclass, field
+from functools import lru_cache
+from typing import Dict, Optional
+
+from opentelemetry import trace as trace_api
+
+from mlflow.tracing.types.model import Status, StatusCode, Trace, TraceData, TraceInfo
+from mlflow.tracing.types.wrapper import MLflowSpanWrapper, NoOpMLflowSpanWrapper
+
+_logger = logging.getLogger(__name__)
+
+
+# Internal representation to keep the state of a trace.
+# Dict[str, Span] is used instead of TraceData to allow access by span_id.
+@dataclass
+class _Trace:
+    trace_info: TraceInfo
+    span_dict: Dict[str, MLflowSpanWrapper] = field(default_factory=dict)
+
+    def to_mlflow_trace(self) -> Trace:
+        span_list = [span.to_mlflow_span() for span in self.span_dict.values()]
```

Comment:
Will we want to return a `sorted(spans, key=lambda span: span.start_time)` collection for TraceData?

Reply:
Theoretically they should be sorted already (cuz they are added to the buffer when created)

Reply:
I was trying to think of a condition where that might not be the case, but I really can't figure out how, even in an async mode that would cause an issue. SGTM - natural insertion order should be totally fine :)